File: api/src/test/java/net/blueshell/api/validation/TodayValidatorTest.java
package net.blueshell.api.validation;

import jakarta.validation.ConstraintValidatorContext;
import net.blueshell.api.config.FactoryConfig;
import net.blueshell.api.config.ValidatorConfig;
import net.blueshell.api.validation.date.TodayValidator;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;

import java.time.LocalDate;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.mockito.Mockito.mock;

@SpringBootTest
class TodayValidatorTest {

    private TodayValidator validator;
    private ConstraintValidatorContext context;

    @BeforeEach
    void setUp() {
        validator = new TodayValidator();
        context = mock(ConstraintValidatorContext.class);
    }

    @Test
    void todayIsValid() {
        assertTrue(validator.isValid(LocalDate.now(), context));
    }

    @Test
    void yesterdayIsInvalid() {
        assertFalse(validator.isValid(LocalDate.now().minusDays(1), context));
    }

    @Test
    void tomorrowIsInvalid() {
        assertFalse(validator.isValid(LocalDate.now().plusDays(1), context));
    }

    @Test
    void nullIsValid() {
        assertTrue(validator.isValid(null, context));
    }
}

File: api/src/test/kotlin/net/blueshell/api/validation/TodayValidatorTest.kt
package net.blueshell.api.validation

import jakarta.validation.ConstraintValidatorContext
import net.blueshell.api.validation.date.TodayValidator
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.kotlin.mock
import org.springframework.boot.test.context.SpringBootTest
import java.time.LocalDate

@SpringBootTest
class TodayValidatorTest {

    private lateinit var validator: TodayValidator
    private lateinit var context: ConstraintValidatorContext

    @BeforeEach
    fun setUp() {
        validator = TodayValidator()
        context = mock()
    }

    @Test
    fun `today is valid`() {
        assertTrue(validator.isValid(LocalDate.now(), context))
    }

    @Test
    fun `yesterday is invalid`() {
        assertFalse(validator.isValid(LocalDate.now().minusDays(1), context))
    }

    @Test
    fun `tomorrow is invalid`() {
        assertFalse(validator.isValid(LocalDate.now().plusDays(1), context))
    }

    @Test
    fun `null is valid`() {
        assertTrue(validator.isValid(null, context))
    }
}
